### ISSUE TITLE
Parallelizer: Spurious parallelization error related to inlined tasks

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#486]
Compiling this program:
```
import "regent"
local __demand(__inline)
task deref( Mesh : region(ispace(int3d),double), elem : int3d(double,Mesh) )
where reads(Mesh) do
    return Mesh[elem]
end
task AdvanceMesh( Mesh : region(ispace(int3d),double) )
where reads writes(Mesh) do
   __demand(__openmp)
   for elem in Mesh do
     for step = 0, 100000 do
       -- this is OK:
       -- Mesh[elem] = Mesh[elem]
       -- this is not OK:
       Mesh[elem] = deref(Mesh, elem)
     end
   end
end
```
produces this error:
```
...lis/sw/legion/language/src/regent/check_parallelizable.t:593: Errors reported during typechecking.
Example.rg:15: OpenMP code generation failed: found a loop-carried dependence
       Mesh[elem] = deref(Mesh, elem)
           ^
```

**Comment by magnatelee** [comment]
Should be fixed by 444462fd.

**Comment by manopapad** [comment]
Fixed
